### ISSUE TITLE
Validation and test set

### ISSUE THREAD

**Issue by Plyb** [DRAGNLabs/MIRROR-Pipeline#141]
**Acceptance Criteria**:
- training can have a validation and test set
- The frequency of validation checks is configurable

Some suggestions:
- A couple options for setting what should be the val/test sets
  - have a totally separate parameter on the `fit` subcommand for each
    - simple
    - need some way of making sure it doesn't overlap with the training set. Easy way to do this to start would be have a `skip` parameter on the dataset constructors in addition to the `head`. The two combined would effectively allow us to set the range when making a dataset. However, this doesn't deal with shuffling. Maybe do shuffling after `skip` and `head`.
  - Rearchitect `MirrorDataset` so that it has separate get_item functions for the train/val/test sets
- Lightning defaults to running the validation set once per epoch, and let's you set the frequency of epochs per val run (so like frequency=0.5 runs a val run every half an epoch). This seems sensible to me, but I don't really have a strong preference for how it works as long as we can set the frequency somehow.

Note: if test sets are adding too much complexity, we can split test and val into two different tickets.
